### PR TITLE
Issue105: Add admin page for Posts & related models

### LINKED ISSUE #102: Admin panel for Posts\Images

### LINKED ISSUE #103: Admin panel for Posts\Authors

### LINKED ISSUE #104: Admin panel for Posts\Labels

### LINKED ISSUE #105: Admin panel for Posts

### PR DESCRIPTION
**Pull request by Sasszem**
If applied, this PR will add backpack admin panels for:
 - `Posts\Labels`
 - `Posts\Authors`
 - `Posts\Images`
 - `Posts\Posts`

Known bugs:
 - #115 
 - #116

Closes #102, closes #103, closes #104, closes #105

### CODE REVIEW

File: app/Http/Requests/AuthorsRequest.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        // only allow updates if the user is logged in
+        return backpack_auth()->check();
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
```
Comment (TwoDCube) [inline app/Http/Requests/AuthorsRequest.php]: Don't forget the validation

File: app/Http/Requests/LabelsRequest.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        // only allow updates if the user is logged in
+        return backpack_auth()->check();
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
```
Comment (TwoDCube) [inline app/Http/Requests/LabelsRequest.php]: Don't forget the validation

File: app/Http/Requests/PostsRequest.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        // only allow updates if the user is logged in
+        return backpack_auth()->check();
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
```
Comment (TwoDCube) [inline app/Http/Requests/PostsRequest.php]: Don't forget the validation

File: app/Models/Posts.php [hunk 4]
```diff
@@ -30,4 +27,76 @@ public function labels()
     {
         return $this->belongsToMany('App\Models\Posts\Labels', 'posts_pivot_labels_data');
     }
+    
+    
+    // @codeCoverageIgnoreStart
```
Comment (TwoDCube) [inline app/Models/Posts.php]: The whole path is ignored if I'm right so we need to delete all codecoverage comments

File: app/Models/Posts.php [hunk 5]
```diff
@@ -10,93 +10,58 @@ class Posts extends Model
...
+            if (count((array)$post->images)) {
+                foreach ($post->images as $file_path) {
+                    \Storage::disk('posts_images')->delete($file_path);
+                }
+            }
+            if (isset($post->index_image)) {
+                \Storage::disk('posts_images')->delete($post->index_image);
+            }
+            
+            $post->author()->associate(null);
```
Comment (TwoDCube) [inline app/Models/Posts.php]: Isn't `dissociate()` a better solution for this?

File: app/Models/Posts/Authors.php [hunk 6]
```diff
@@ -15,80 +16,31 @@ class Authors extends Model
...
-        $this->append('image');
-        return parent::toJson($options);
+        parent::boot();
+        static::deleting(function (Authors $author) {
+            if (isset($author->image)) {
+                \Storage::disk('authors_images')->detach($author->image);
+            }
+
+            Posts::where('author_id', '=', $author->id)->get()->each(function (Posts $p) {
+                $p->author()->associate(null);
```
Comment (TwoDCube) [inline app/Models/Posts/Authors.php]: Isn't `dissociate()` a better solution for this?

File: app/Models/Posts/Authors.php [hunk 7]
```diff
@@ -15,80 +16,31 @@ class Authors extends Model
...
-    
-    // 'Custom' serializer to use accessor for image attribute
-    public function toJson($options=0)
-    {
-        $this->append('image');
-        return parent::toJson($options);
+        parent::boot();
+        static::deleting(function (Authors $author) {
+            if (isset($author->image)) {
+                \Storage::disk('authors_images')->detach($author->image);
```
Comment (TwoDCube) [inline app/Models/Posts/Authors.php]: Why is this `detach` and not `delete`?

File: app/Models/Posts.php (hunk 5 above)
Comment (Sasszem) [reply app/Models/Posts.php]: Fixed

File: app/Models/Posts/Authors.php (hunk 6 above)
Comment (Sasszem) [reply app/Models/Posts/Authors.php]: Fixed

File: app/Models/Posts/Authors.php (hunk 7 above)
Comment (Sasszem) [reply app/Models/Posts/Authors.php]: Changed to delete

**Pull request merged**